roles/kubernetes-apps/ansible/tasks/main.yml

```diff
@@ -11,68 +11,17 @@
   delay: 2
   when: inventory_hostname == groups['kube-master'][0]
 
-- name: Kubernetes Apps | Delete old kubedns resources
-  kube:
-    name: "kubedns"
-    namespace: "{{ system_namespace }}"
-    kubectl: "{{ bin_dir }}/kubectl"
-    resource: "{{ item }}"
-    state: absent
-  with_items:
-    - 'deploy'
-    - 'svc'
+- name: Kubernetes Apps | KubeDNS
+  import_tasks: tasks/kubedns.yml
```

Comment:
better is to use include:
include: tasks/{{ dns_mode }}.yml
when: 
  - "dns_mode != 'nonce'"
  - "dns_mode in ['kubedns', 'coredns' ]
this way you need to write this once :)

Reply:
k8s-cluster.yml file says dns mode can be:

```
# Can be dnsmasq_kubedns, kubedns, manual or none
# dns_mode: kubedns
```

There is no `dnsmasq_kubedns` file.

Reply:
ok, then it makes sense

Reply:
include: tasks/{{ dns_mode }}.yml
when:

"dns_mode != 'nonce'"
"dns_mode in ['kubedns', 'coredns' ]

I also agree with this writing
Otherwise ，Failing to deploy kubedns.

The 'manifests' variable is defined twice

Reply:
What about now?